src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Couldn't create an order: " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getAllByUser(User user) {
```

Comment:
it would be better to rename it 
```suggestion
    public List<Order> getOrdersHistory(User user) {
```